### PR TITLE
Feature/kas 2303 custom subcase service to resource calls

### PR DESCRIPTION
**Pull request by ValenberghsSven**
https://kanselarij.atlassian.net/browse/KAS-2303

Already pushed the **backend** changes to existing backend branch by mistake :( 
App: https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/315

PR#1483 and PR#1502 can run against development backend so should not be much of a problem.
This PR can also run against the development backend and shouldn't need a custom jenkins build.

refactor create-agendaitem component:
- stopped using `<DataTable>`, was having issues with pagination in octane
- removed "show postponed" feature, didn't work anymore, too complex to calculate for all postponed subcases at once with recent changes.

fixed typos in `confirmationModal`

subcase-description-view component:
- use new `Subcase::Timeline` component

subcase-item component:
- use new `Subcase::Timeline` component

new `Subcase::Timeline` component to replace custom-subcase-service call

remove subcase-service service.

Cleanup tests and unused translations

### CODE REVIEW

File: app/components/agenda/agendaitem/create-agendaitem.hbs [hunk 1]
```diff
@@ -24,77 +14,99 @@
...
                   @icon="search"
                   @block={{true}}
-                  @value={{this.filterParam}}
+                  @value={{this.filter}}
                   id="searchId"
                   placeholder={{t "search-placeholder"}}
-                  disabled={{this.showPostponed}}
                   data-test-create-agendaitem-input={{true}}
-                  {{on "input" (perform this.searchTask)}}
+                  {{on "input" this.searchTask.perform}}
```
Comment (erikap) [inline app/components/agenda/agendaitem/create-agendaitem.hbs]: Why not using `(perform this.searchTask)`?

File: app/components/agenda/agendaitem/create-agendaitem.hbs [hunk 2]
```diff
@@ -24,77 +14,99 @@
...
\ No newline at end of file
+    {{/if}}
+  </Auk::Modal::Body>
+  <Auk::Modal::Footer @onCancel={{@onClose}}>
+    <AuButton
+      data-test-create-agendaitem-save
+      @skin="primary"
+      @loading={{this.addSubcasesToAgenda.isRunning}}
+      @disabled={{this.isSaveDisabled}}
+      {{on "click" this.addSubcasesToAgenda.perform}}
```
Comment (erikap) [inline app/components/agenda/agendaitem/create-agendaitem.hbs]: Use `(perform this.addSubcasesToAgenda)`

File: app/components/agenda/agendaitem/create-agendaitem.js [hunk 3]
```diff
@@ -1,255 +1,137 @@
...
-  },
-});
+  @action
+  onSortChange(sortField) {
+    this.sort = sortField;
+    this.findAll.perform();
+  }
+
+  @action
+  async selectSubcase(subcase) {
```
Comment (erikap) [inline app/components/agenda/agendaitem/create-agendaitem.js]: Remove `async`

File: app/components/subcase/timeline.hbs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        class="vlc-status-timeline__item vlc-status-timeline__item--neutral"
+        data-test-subcase-timeline-item
+      >
+        <div class="vlc-status-timeline__dot"></div>
+        {{#if (is-pending phase)}}
+          <div
+            class="auk-skeleton auk-skeleton--rectangle"
+            role="alert"
+            aria-busy="true"
+          ></div>
```
Comment (erikap) [inline app/components/subcase/timeline.hbs]: Phases are just plain objects, no promises. If branch will never render?

**Review by erikap — COMMENTED** [review]
Some small comments, but over all major improvement of the code :+1:

File: app/components/agenda/agendaitem/create-agendaitem.hbs (hunk 1 above)
Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/create-agendaitem.hbs]: not sure why ...:brain: 
will change

File: app/components/agenda/agendaitem/create-agendaitem.hbs (hunk 2 above)
Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/create-agendaitem.hbs]: will change

File: app/components/subcase/timeline.hbs (hunk 4 above)
Comment (ValenberghsSven) [reply app/components/subcase/timeline.hbs]: glossed over this after copy paste

File: app/components/agenda/agendaitem/create-agendaitem.js (hunk 3 above)
Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/create-agendaitem.js]: :+1:

**Comment by ValenberghsSven** [comment]
only 2 failed tests, 1 flaky (concepts) and 1 time based test that is only failing today (wrong month expected). that one is failing on all branches.
Merging this one already

**Pull request merged**